Lib/types.py

```diff
@@ -143,6 +143,34 @@ def _calculate_meta(meta, bases):
                         "of the metaclasses of all its bases")
     return winner
 
+
+def get_original_bases(cls, /):
+    """Return the class's "original" bases prior to modification by `__mro_entries__`.
+
+    This is useful for introspecting Generics.
+
+    Examples::
+
+        from typing import TypeVar, Generic
+
+        T = TypeVar("T")
+        class Foo(Generic[T]): ...
+        class Bar(Foo[int], float): ...
+        class Baz(list[str]): ...
+
+        get_original_bases(Foo) == (Generic[T],)
+        get_original_bases(Bar) == (Foo[int], float)
+        get_original_bases(Baz) == (list[str],)
+        get_original_bases(int) == None
+    """
+    if isinstance(cls, type):
+        try:
+            return cls.__orig_bases__
+        except AttributeError:
+            pass
+    return None
```

Comment:
why not `return ()` / empty tuple ? eventually only for/instead of the `pass`  statement ?

Reply:
I think I prefer Alex's suggestion more here